packages/quantic/force-app/main/default/lwc/quanticFacet/__tests__/quanticFacet.test.js

```diff
@@ -231,9 +333,674 @@ describe('c-quantic-facet', () => {
+                  highlightedResult: facetValue.value,
+                })
+              );
+              const element = createTestComponent();
+              await flushPromises();
+
+              const facetValueElements = element.shadowRoot.querySelectorAll(
+                selectors.facetValue
+              );
+              expect(facetValueElements.length).toBe(
+                expectedFacetValues.length
+              );
+              expectedFacetValues.forEach((facetValue, index) => {
+                expect(facetValueElements[index].item).toEqual(facetValue);
+              });
+            });
+          });
+
+          describe('when the facet search input is not empty', () => {
+            it('should display the facet values as facet search values', async () => {
+              const expectedFacetValues = exampleFacetSearchValues.map(
+                (facetValue) => ({
+                  checked: false,
+                  highlightedResult: facetValue.displayValue,
+                  value: facetValue.rawValue,
+                  numberOfResults: facetValue.count,
+                  state: 'idle',
+                })
+              );
+              const element = createTestComponent();
+              await flushPromises();
+
+              const searchboxInput = element.shadowRoot.querySelector(
+                selectors.searchboxInput
+              );
+              searchboxInput.value = 'foo';
+              await flushPromises();
+
+              const facetValueElements = element.shadowRoot.querySelectorAll(
+                selectors.facetValue
+              );
+              expect(facetValueElements.length).toBe(
+                exampleFacetSearchValues.length
+              );
+              expectedFacetValues.forEach((facetValue, index) => {
+                expect(facetValueElements[index].item).toEqual(facetValue);
+              });
+            });
+
+            ['checkbox', 'link'].forEach((propertyValue) => {
+              describe(`when the property displayValuesAs is set to "${propertyValue}"`, () => {
+                it(`should display the facet search value as ${propertyValue}`, async () => {
+                  const element = createTestComponent({
+                    ...defaultOptions,
+                    displayValuesAs: propertyValue,
+                  });
+                  await flushPromises();
+
+                  const searchboxInput = element.shadowRoot.querySelector(
+                    selectors.searchboxInput
+                  );
+                  searchboxInput.value = 'foo';
+                  await flushPromises();
+
+                  const facetValueElement = element.shadowRoot.querySelector(
+                    selectors.facetValue
+                  );
+                  expect(facetValueElement.displayAsLink).toEqual(
+                    propertyValue === 'link'
+                  );
+                });
+
+                const expectedFunctionToBeCalled =
```

Comment:
This you could make it better either in your foreach or in a describe.each though :) 

`[{propertyValue: 'checkbox', functionToCall: 'select'}, {propertyValue: 'link', functionToCall: 'singleSelect'}].forEach...`

or 

```
describe.each([
  {propertyValue: 'checkbox', functionToCall: 'select', ...},
  {propertyValue: 'link', functionToCall: 'singleSelect', ...},
])('the property displayValuesAs', ({propertyValue, functionToCall, ...}) => {
```